docs/developers/decisions/backend/0014-loadout-items.md

```diff
@@ -0,0 +1,120 @@
+# Loadout Items
+
+## Context and Problem Statement
+
+Currently, the app follows the standard mod manager approach of loadouts being a tree of `loadout -> mods -> files`
+
+## The Current Situation
+The current approach is mostly driven by tradition, the fact that other mod managers do it this way, mostly due to the need
+keep files on disk extracted so they can be linked into the game folder or easily accessable by VFS code. This is not a hard
+requirement for us, and we can do things differently
+
+To start this design process, let's enumerate the sorts of things we would want to store in a loadout:
+
+- Mods downloaded from the internet
+- Config files for these mods
+- FOMOD or installer option selections
+- 'Stored files' or files that are used verbatim by the game, like textures or meshes
+- Generated files, like merged patches or bashed patches
+- Game settings, like INI files
+- Nexus Collections installed by the user. 
+
+## The Solution
+
+The decision made (mostly through online discussions) is to store all of these items in a single parent model in MnemonicDB:
+this will be called a `loadout item` and the specific behavior of each item will be based on the polymorphic nature
+of these models. To start, we will design the fairly simple data model:
+
+```mermaid
+
+classDiagram
+    class LoadoutItem {
+        +EntityId Loadout
+        +EntityId? Parent
+    }
+    
+
+    class File {
+        +GamePath To
+    }
+
+    class StoredFile {
+        +Hash Hash
+        +Size Size
+    }
+
+    class GeneratedFile {
+        +Guid GeneratorId
+    }
+    
+    class ContainerItem {
+        +string Name
+        +marker IsDisabled
+    }
+    
+    class DownloadedMod {
+        +EntityId
+        +ReferenceId DownloadMetadata
+    }
+    
+    class GameFiles {
+         +GameInstallation Installation
+    }
+
+    LoadoutItem <|-- File
+    File <|-- StoredFile
+    File <|-- GeneratedFile
```

Comment:
I'm not sure I understood this part:
![image](https://github.com/Nexus-Mods/NexusMods.App/assets/26797547/6c5caac6-209d-4c5e-91bc-e97a85df6647)

My thought was that we would have an another level between `LoadoutItem` and the Files, something like `FileBucket`. 

The idea being that not all `LoadoutItems` would necessarily have files attached to them. If it is a `FileBucket` then it would have files and it would also have any extra data or functionality that might be needed for Files.

If a particular LoadoutItem is not supposed to have Files, then it would not be a `FileBucket` or whatever name we choose, but it would be something else, like `ModGroup` or `Ini Edit`.

Maybe the arrows were supposed to represent a different kind of relationship with `LoadoutItem` than what expressed with `GameFiles` and `ContainerItem`? If they are meant to be the same `Include<LoadoutItem>` kind of relationship, then that confuses me.

Reply:
The arrows represent a "IsA<Parent>" relationship, and that's not really clear in that graph. A StoredFile `isa` File which `isa` LoadoutItem. What you're suggesting is what I called `ContainerItem` which is something that groups files in some way, like a mod, game files container, etc.

Reply:
This still seems a bit confusing to me, but maybe the idea is for Files to be LoadoutItems but they would usually have a parent that would be a File container.

I'm not sure we would ever have top level File LoadoutItems, as there would always need to be some context information attached to it, that we wouldn't want to have on all File Items.

I had not grasped from the original idea that LoadoutItem would scale down to File level.
I guess I'm not sure whether this would actually be beneficial as opposed to having a  FileBucket type entity that would represent a collection of Files, with any necessary metadata related to this collection of files.

I can see the symmetry that spawned the idea to just expand it lower, but I'm not sure whether doing so actually has tangible benefits other than having a system fully defined within itself. I found the concept a bit confusing, and I don't see Files being used directly without a dedicated FileBucket type container. 
I will read this through again tomorrow to see if I missed things.